### PR TITLE
Added the interaction feature

### PR DESCRIPTION
**Pull request by Geleta116**
I have implemented the interaction flow from domain to web api. I have used Unit of work for this.

### CODE REVIEW

File: aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs [hunk 1]
```diff
@@ -2,4 +2,8 @@
 
 namespace SocialSync.Application.Contracts.Persistence;
 
-public interface IInteractionRepository : IGenericRepository<Interaction> { }
+public interface IInteractionRepository : IGenericRepository<Interaction> {  
+     Task<List<Interaction>> GetAllCommentInteractions(int PostId);
```
Comment (bahailu-abera) [inline aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs:6]: Prefer to use Async suffix for Async method names(for readability and consistency).

File: aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs [hunk 2]
```diff
@@ -2,4 +2,8 @@
...
+
+    Task<Interaction> likeUnlikeInteraction(Interaction interaction);
```
Comment (bahailu-abera) [inline aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs:8]: use Pascal case for method names(this rule applies for all methods).

File: aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs (hunk 1 above)
Comment (Geleta116) [reply aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs]: I have resolved it.

[the remark this answers is not in the document]
Comment (Geleta116) [reply aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs]: I have resolved it.

File: aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs (hunk 2 above)
Comment (Geleta116) [reply aait/backend/group-2B/Application/Contracts/Persistence/IInteractionRepository.cs]: I have resolved it.

**Pull request closed without merging**